xxxx-gpu-msm.md

```diff
@@ -0,0 +1,92 @@
+# RFC title
+
+GPU based MSM implementation.
+
+## Summary
+
+A rust library implementing gpu based acceleration for computing MSM, it can be integrated in kimchi and
+other projects that make use of MSMs with the pasta curves.
+
+## Motivation
+
+One of the mains components of kimchi's runtime is the multi scalar multiplication (MSM), it consist of
+basically taking a list of points, adding each point to itself N times, and then add all the resulting
+points together.
+This library will accelerate that process doing the computation in a gpu, it will also serve as a reference
+point to measure the improvements that gpu computations can provide us.
+
+This will build on top of an already existing implementation, and thus some of the design choices will be
+based on the fact that the particular decision is cheap to implement because it is already implemented.
+
+## Detailed design
+
+The implementation will be based in the wgpu graphics API and thus the WGSL shader language. The library
+will consist of 3 crates:
+    - arithmetic
+    - shader
+    - runner
+
+### arithmetic
+
+This is a crate implementing general field arithmetic, general curve arithmetic and specifically the
+pasta curves and their field. It uses no dependencies and avoids hardware features that are unsupported
+by GPUs in general and WGSL in particular.
+The code is also written in a way optimized for GPUs, but it can still be used in any architecture, while
+it will be generally slower that other cpu implementations it can be used for things like tests.
+
+### shader
+
+The shader is a small crate that will be compiled to WGSL, the code is very simple, mostly calls to
+functions from the arithmetic crate, compilation takes a significative time and thus the compiled
+shader will be saved and the runner will used that precompiled shader by default.
+
+### runner
+
+This is library that will be used by end users of the crate, it is the part that will run in the CPU
+and will talk to the GPU, it will expose a simple API that will ask for the points that make the base
```

Comment:
Can you add a tiny sketch of the API inline?

Reply:
added